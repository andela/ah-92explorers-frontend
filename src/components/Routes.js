/* eslint-disable import/no-unresolved */
import React, { Fragment } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import CreateArticle from './Articles/CreateArticle.jsx';
import UpdateArticle from './Articles/UpdateArticle.jsx';
import { ReadArticle } from './Articles/ArticleItem.jsx';
import Signup from './Auth/Signup.jsx';
import Feed from './Articles/Feed.jsx';
import Login from './Containers/Login.jsx';
import ViewProfile from './Profile/ViewProfile';
import EditProfile from './Profile/EditProfile';
import ResetPassword from './Auth/ResetPassword';
import ResetingPassword from './Auth/ResettingPassword';
import ViewArticleRatings from './Articles/getRatings';
import Follow from './Follow/Follow.jsx';
import { NotFound } from './NotFound/NotFound';

const Routes = () => (
  <Router>
    <Switch>
      <Fragment>
        <Route exact path="/article" component={CreateArticle} />
        <Route exact path="/article/:articleSlug/update" component={UpdateArticle} />
        <Route exact path="/article/:articleSlug" component={ReadArticle} />
        <Route exact path="/rating/:articleSlug" component={ViewArticleRatings} />
        <Route exact path="/" component={Feed} />
        <Route path="/signup" component={Signup} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/profile" component={ViewProfile} />
        <Route exact path="/user-profile/:username" component={Follow} />
        <Route exact path="/profile/edit" component={EditProfile} />
        <Route exact path="/reset-password" component={ResetPassword} />
        <Route exact path="/resetting-password" component={ResetingPassword} />
        <Route exact path="/not-found" component={NotFound} />
      </Fragment>
    </Switch>
  </Router>
);


export default Routes;
