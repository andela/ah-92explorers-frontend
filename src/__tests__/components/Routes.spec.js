import React from 'react';
import { shallow } from 'enzyme';
import Routes from '../../components/Routes';

describe('<Routes />', () => {
  it('should render App component', () => {
    const component = shallow(<Routes />)
    expect(component).toMatchSnapshot();
  });
});
