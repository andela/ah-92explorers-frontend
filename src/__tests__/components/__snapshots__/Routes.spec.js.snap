// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should render App component 1`] = `ShallowWrapper {}`;
